sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -461,6 +466,34 @@ static ssize_t _write_options(coap_pkt_t *pdu, uint8_t *buf, size_t len)
     return bufpos - buf;
 }
 
+/*
+ * Compares the addresses and ports for two sock UDP endpoints.
+ *
+ * ep1[in] -- endpoint
+ * ep2[in] -- endpoint
+ *
+ * return Result of memcmp (-1, 0, 1) if address family and port match;
+ *        otherwise -2.
+ */
+static int _cmp_endpoints(const sock_udp_ep_t *ep1, const sock_udp_ep_t *ep2)
```

Comment:
from its usage, IMHO a `static bool _endpoints_equal(ep1, ep2)` function would be more appropriate, which returns only true or false. 

For example (untested):
```C
static bool _endpoints_equal(const sock_udp_ep_t *ep1, const sock_udp_ep_t *ep2)
{
    if (ep1->family != ep2->family) {
        return false;
    }
    if (ep1->port != ep2->port) {
        return false;
    }
    if ((ep1->family == AF_INET) && (ep1->addr.ipv4_u32 != ep2->addr.ipv4_u32)) {
        return false;
    }
    else if (memcmp(&ep1->addr.ipv6[0], &ep2->addr.ipv6[0], 16) != 0) {
        return false;
    }
    return true;
} 
```

Only a suggestion, what do you think? Otherwise I trust that the changes fit your grander scheme 😉

Reply:
Thanks, @smlng! I agree that your function fits the purpose better. Working on fixup commit.